### PR TITLE
Add initial BYOB-based SLSA-generator

### PR DESCRIPTION
**Pull request by AdamKorcz**
#### Summary

Adds the initial workflow for the Bring-Your-Own-Builder-based SLSA generator. 

Currently the builder is hosted on my Github user, because the builder is under development. I am uploading it here to get it integrated early in the development process of the gradle builder. The BYOB-based generator should not disrupt any other workflows since it it needs to be manually triggered to run.

### CODE REVIEW

**Comment by loosebazooka** [comment]
Fyi. I'm working with Adam on this.

**Review by loosebazooka — APPROVED** [review]

File: .github/workflows/byob-slsa.yaml [hunk 1]
```diff
@@ -0,0 +1,28 @@
+# This builds a SLSA provenance statement based on BYOB.
+# For now it is under heavy development and is not yes suited for releases.
```
Comment (loosebazooka) [inline .github/workflows/byob-slsa.yaml:2]:
```suggestion
# For now it is under heavy development and is not yet suited for releases.
```

File: .github/workflows/byob-slsa.yaml [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+    permissions:
+      contents: write
+      id-token: write
+      actions: read
+      packages: write
+    uses: AdamKorcz/java-slsa-generator/.github/workflows/gradle-trw.yml@main
+    with:
+      rekor-log-public: true
+      artifact-list: |
+        ./sigstore-java/build/local-maven-repo/dev/sigstore/sigstore-java/GRADLE_VERSION/sigstore-java-GRADLE_VERSION.module,
```
Comment (loosebazooka) [inline .github/workflows/byob-slsa.yaml:24]: the documentation says this should be GRADLEVERSION? https://github.com/AdamKorcz/java-slsa-generator/blob/main/.github/workflows/gradle-trw.yml#L28

**Review by loosebazooka — COMMENTED** [review]
What's the testing plan here? Are we running this on a tag seeing what happens?

File: .github/workflows/byob-slsa.yaml (hunk 2 above)
Comment (AdamKorcz) [reply .github/workflows/byob-slsa.yaml:24]: Was a mistake in the documentation.

**Comment by AdamKorcz** [comment]
Just ran another build which succeeded: https://github.com/AdamKorcz/sigstore-java/actions/runs/4609029248

**Comment by AdamKorcz** [comment]
> What's the testing plan here? Are we running this on a tag seeing what happens?

Yes. Let's run it paralel to the existing build workflow.

**Review by loosebazooka — APPROVED** [review]

**Pull request merged**